onchain/rollups/contracts/consensus/authority/AuthorityHistoryPairFactory.sol

```diff
@@ -0,0 +1,109 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+pragma solidity ^0.8.8;
+
+import {IAuthorityHistoryPairFactory} from "./IAuthorityHistoryPairFactory.sol";
+import {Authority} from "./Authority.sol";
+import {IAuthorityFactory} from "./IAuthorityFactory.sol";
+import {History} from "../../history/History.sol";
+import {IHistoryFactory} from "../../history/IHistoryFactory.sol";
+
+/// @title Authority-History Pair Factory
+/// @notice Allows anyone to reliably deploy a new Authority-History pair.
+contract AuthorityHistoryPairFactory is IAuthorityHistoryPairFactory {
+    IAuthorityFactory immutable authorityFactory;
+    IHistoryFactory immutable historyFactory;
+
+    /// @notice Constructs the factory.
+    /// @param _authorityFactory The `Authority` factory
+    /// @param _historyFactory The `History` factory
+    constructor(
+        IAuthorityFactory _authorityFactory,
+        IHistoryFactory _historyFactory
+    ) {
+        authorityFactory = _authorityFactory;
+        historyFactory = _historyFactory;
+
+        emit AuthorityHistoryPairFactoryCreated(
+            _authorityFactory,
+            _historyFactory
+        );
+    }
+
+    function getAuthorityFactory()
+        external
+        view
+        override
+        returns (IAuthorityFactory)
+    {
+        return authorityFactory;
+    }
+
+    function getHistoryFactory()
+        external
+        view
+        override
+        returns (IHistoryFactory)
+    {
+        return historyFactory;
+    }
+
+    function newAuthorityHistoryPair(
+        address _authorityOwner
+    ) external override returns (Authority authority_, History history_) {
+        authority_ = authorityFactory.newAuthority(address(this));
+        history_ = historyFactory.newHistory(address(authority_));
+
+        authority_.setHistory(history_);
+        authority_.transferOwnership(_authorityOwner);
```

Comment:
Doesnt the Authority contract already transfer ownership to the constructor parameter address in the contructor? I dont think we need this step if we do `authorityFactory.newAuthority(_authorityOwner);`

Reply:
We need the `AuthorityHistoryPairFactory` to be the owner of the newly-created `Authority` in order to set its `History`.